sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -42,6 +42,12 @@ class BooleanSimplificationSuite extends PlanTest with PredicateHelper {
 
   val testRelation = LocalRelation('a.int, 'b.int, 'c.int, 'd.string)
```

Comment:
How about adding another boolean column?
```
  val testRelation = LocalRelation('a.int, 'b.int, 'c.int, 'd.string, 'e.boolean)
```

Our filter conditions are not allowed to accept the non-boolean predicates